tests/UI/Component/Panel/PanelTest.php

```diff
@@ -0,0 +1,241 @@
+		$sub = $f->sub("Title",array(new ComponentDummy()));
+		$p = $f->report("Title",array($sub));
+
+		$p = $p->withTitle("TitleNew");
+
+		$this->assertEquals($p->getTitle(), "TitleNew");
+	}
+
+	public function test_report_get_content() {
+		$f = $this->getPanelFactory();
+		$sub = $f->sub("Title",array(new ComponentDummy()));
+		$p = $f->report("Title",$sub);
+
+		$this->assertEquals($p->getSubPanels(), array($sub));
+	}
+
+	public function test_report_with_sub_panel() {
+		$f = $this->getPanelFactory();
+		$sub1 = $f->sub("Title",array(new ComponentDummy(1)));
+		$p = $f->report("Title",array($sub1));
+
+		$sub2 = $f->sub("Title",array(new ComponentDummy()));
+		$p = $p->withSubPanels($sub2);
+
+		$this->assertEquals($p->getSubPanels(), array($sub2));
+	}
+
+	public function test_render_standard() {
+		$f = $this->getPanelFactory();
+		$r = $this->getDefaultRenderer();
+		$p = $f->standard("Title",array());
+
+		$html = new DOMDocument();
+		$html->formatOutput = true;
+		$html->preserveWhiteSpace = false;
+
+		$expected = new DOMDocument();
+		$expected->formatOutput = true;
+		$expected->preserveWhiteSpace = false;
+
+		$html->loadXML($this->normalizeHTML($r->render($p)));
+
+
+		$expected->loadXML($this->normalizeHTML(
+				"<div class=\"panel panel-primary\">".
+				"   <div class=\"panel-heading ilHeader\">".
+				"       <h3 class=\"ilHeader\">Title</h3>".
+				"   </div>".
+				"   <div class=\"panel-body\"></div>".
+				"</div>"));
+
+		$this->assertEquals($expected->saveHTML(), $html->saveHTML());
+	}
+
+	public function test_render_sub() {
+		$f = $this->getFactory();
+		$fp = $this->getPanelFactory();
+		$r = $this->getDefaultRenderer();
+		$p = $fp->sub("Title",array());
+		$card = $f->card("Card Title");
+		$p = $p->withCard($card);
+
+		$html = new DOMDocument();
+		$html->formatOutput = true;
+		$html->preserveWhiteSpace = false;
+
+		$expected = new DOMDocument();
+		$expected->formatOutput = true;
+		$expected->preserveWhiteSpace = false;
+
+		$html->loadXML($this->normalizeHTML($r->render($p)));
+
+
+		$expected->loadXML($this->normalizeHTML(
+				"<div class=\"panel panel-default\">".
+				"   <div class=\"panel-heading ilBlockHeader\">".
+				"       <h3 class=\"ilBlockHeader\">Title</h3>".
+				"   </div>".
+				"   <div class=\"panel-body\"><div class=\"row\">".
+				"       <div class=\"col-sm-8\"></div>".
+				"       <div class=\"col-sm-4\">".
+				"           <div class=\"il-card thumbnail\"><div class=\"caption\"><h2 class=\"card-title\">Card Title</h2></div></div>".
+				"       </div>".
+				"   </div></div>".
+				"</div>"));
+
+		$this->assertEquals($expected->saveHTML(), $html->saveHTML());
+	}
+	public function test_render_report() {
+		$f = $this->getFactory();
+		$fp = $this->getPanelFactory();
+		$r = $this->getDefaultRenderer();
+		$sub = $fp->sub("Title",array());
+		$card = $f->card("Card Title");
+		$sub = $sub->withCard($card);
+
+		$report = $fp->report("Title",$sub);
+		$html = new DOMDocument();
```

Comment:
This pattern looks nice. Maybe we could abstract it a little and promote it to the base class for tests?